session.go

```diff
@@ -57,9 +58,19 @@ type Session interface {
 	// of whether or not a PTY was accepted for this session.
 	Pty() (Pty, <-chan Window, bool)
 
-	// TODO: Signals(c chan<- Signal)
+	// Signals registers a channel to receive signals sent from the client. The
+	// channel must handle signal sends or it will block the SSH request loop.
+	// Registering nil will unregister the channel from signal sends. During the
+	// time no channel is registered signals are buffered up to a reasonable amount.
+	// If there are buffered signals when a channel is registered, they will be
+	// sent in order on the channel immediately after registering.
+	Signals(c chan<- Signal)
```

Comment:
Personally I really don't see why we should. It's a lot more management of something that isn't even used that much. OpenSSH doesn't even handle signals. I only today realized that most signals via PTY are sent via escape codes, so the majority of the common signal cases are already handled.